pkg/provision/source_test.go

```diff
@@ -5,36 +5,72 @@ import (
 	"testing"
 	"time"
 
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
 	"github.com/threefoldtech/zos/pkg"
 )
 
-type testReservationStore struct {
-	fReserve func(r Reservation, nodeID pkg.Identifier) error
-	fPoll    func(nodeID pkg.Identifier, all bool, since time.Time) ([]*Reservation, error)
-	fGet     func(id string) (Reservation, error)
+type TestPollSource struct {
+	mock.Mock
 }
 
-func (s testReservationStore) Reserve(r Reservation, nodeID pkg.Identifier) error {
-	return s.fReserve(r, nodeID)
-}
-func (s testReservationStore) Poll(nodeID pkg.Identifier, all bool, since time.Time) ([]*Reservation, error) {
-	return s.fPoll(nodeID, all, since)
-}
-func (s testReservationStore) Get(id string) (Reservation, error) {
-	return s.fGet(id)
+func (s *TestPollSource) Poll(nodeID pkg.Identifier, all bool, since time.Time) ([]*Reservation, error) {
+	returns := s.Called(nodeID, all, since)
+	return returns.Get(0).([]*Reservation), returns.Error(1)
 }
 
 func TestHTTPReservationSource(t *testing.T) {
-	t.Skip()
-	store := testReservationStore{
-		fPoll: func(nodeID pkg.Identifier, all bool, since time.Time) ([]*Reservation, error) {
-			time.Sleep(time.Second * 10)
-			return []*Reservation{}, nil
-		},
+	require := require.New(t)
+	var store TestPollSource
+
+	nodeID := pkg.StrIdentifier("node-id")
+	source := PollSource(&store, nodeID)
+	chn := source.Reservations(context.Background())
+
+	store.On("Poll", nodeID, true, mock.Anything).
+		Return([]*Reservation{
+			&Reservation{ID: "res-1"},
+			&Reservation{ID: "res-2"},
+		}, ErrPollEOS)
+
+	reservations := []*Reservation{}
+	for res := range chn {
+		reservations = append(reservations, res)
+	}
+
+	require.Len(reservations, 2)
+	require.Equal("res-1", reservations[0].ID)
+}
```

Comment:
```suggestion
require.Equal("res-2", reservations[1].ID)
```
Let's test them all since they are only 2 results